packages/cli/src/services/bucket/base.ts

```diff
@@ -24,20 +26,28 @@ export abstract class BaseBucketProcessor implements IBucketProcessor {
     return result;
   }
 
-  async translate(payload: BucketPayload, sourceLocale: string, targetLocale: string): Promise<BucketPayload> {
+  async translate(payload: BucketPayload, sourceLocale: string, targetLocale: string, onProgress?: (progress: number) => void): Promise<BucketPayload> {
     // The data contains key-value pairs, so let's translate
-    // the values in batches of 20 keys max.
     const resultData: Record<string, any> = {};
 
     const keys = Object.keys(payload.data);
-    const batches = _.chunk(keys, 20);
+    const batches = _.chunk(keys, CHUNK_SIZE);
+    const totalBatches = batches.length;
+
+    let completedBatches = 0;
+
     for (const batch of batches) {
       const partialData = _.pick(payload.data, batch);
       const partialPayload = { data: partialData, meta: payload.meta };
       const partialResult = await this.translator(sourceLocale, targetLocale, partialPayload);
       _.merge(resultData, partialResult);
+      completedBatches++;
+
+      if (onProgress) {
+        const progress = (completedBatches / totalBatches) * 100;
+        onProgress(progress);
+      }
```

Comment:
Actually you can even change it to

```suggestion
      const progress = (completedBatches / totalBatches) * 100;
      onProgress?.(progress);
```